src/libfuncs/cast.rs

```diff
@@ -118,39 +111,34 @@ pub fn build_downcast<'ctx, 'this>(
                 ),
             )
             .ok_or(Error::ParseAttributeError)?;
-            let half_prime: melior::ir::Value = block
-                .append_operation(arith::constant(context, attr_halfprime_i252, location))
-                .result(0)?
-                .into();
+            let half_prime = block
+                .append_op_result(arith::constant(context, attr_halfprime_i252, location))?;
 
             let is_felt_neg = block
-                .append_operation(arith::cmpi(
+                .append_op_result(arith::cmpi(
                     context,
                     CmpiPredicate::Ugt,
                     src_value,
                     half_prime,
                     location,
-                ))
-                .result(0)?
-                .into();
-
+                ))?;
             let is_neg_block = helper.append_block(Block::new(&[]));
             let is_not_neg_block = helper.append_block(Block::new(&[]));
             let final_block = helper.append_block(Block::new(&[(src_ty, location)]));
 
-            block.append_operation(cf::cond_br(
```

Comment:
we dont need the result (in fact, it may have no result and this would panic maybe), so it was correct how it was before.